lib/interface.js

```diff
@@ -0,0 +1,64 @@
+$(document).ready(function () {
+
+  game = new Game();
+  game.addFrame(new Frame());
+
+  $('.hit-pins').on('click', function () {
+    addRoll(parseInt($(this).val()))
+  });
+
+  function addRoll(pins) {
+    game.addRoll(pins)
+    hidePinButtons(pins)
+    updateTotalScore()
+    writeRolls()
+    writeFrames()
+  }
+
+  function updateTotalScore() {
+    $("#current-score").text(game.score());
+  };
+
+  function writeRolls() {
+    for (n = 0; n < game.currentFrameNo(); n++) {
+      var frame = game.getFrames()[n]
+      $("#frame" + (n + 1) + "roll1").text(getFirstRoll(frame))
```

Comment:
Rather than giving everything ids like 'frame1' and 'frame2', what if you conceived of your frame interface as a component with it's own class? So then rather than a table, you end up with something like:

```
function FrameView() { }

FrameView.prototype = {
  initialise: function() {
    this._element = $(this.frameViewElement).html(
      '<div class='frame-view'>' +
        '<div class='frame-number'></div>' +
        '<div class='frame-roll1'></div>' +
        '<div class='frame-roll2'></div>' +
      '</div>'
    );
  },

  appendTo: function(parent) {
    $(parent).append(this._element);
  }

  recordRolls: function(frame) {
    $(element).find('.frame-roll1').text(getFirstRoll(frame));
    $(element).find('.frame-roll2').text(getSecondRoll(frame));
  }
}

var parentElement = $('#score-card');
var frameViews = [];

for (var i = 0; i < MAX_FRAMES; i++) {
  var frameView = new FrameView();
  frameView.initialise();
  frameView.appendTo(parentElement);
  frameViews.push(frameView);
}

...

for (n = 0; n < game.currentFrameNo(); n++) {
  frameViews.setScores(get.getFrames()[n]);
}

```

I've missed some code above and I'm sure I've made a mistake or two typing it out, but the point is to demonstrate that UI components can also be considered in an OO way and this can help you reduce repetition and encapsulate some of your logic, which will scale better when building more complex UIs.